### PR TITLE
Match GlobeView projection parameters with Maplibre v5

### PR DESCRIPTION
**Pull request by Pessimistress**
For #9199 

globe

Grid lines are rendered by Maplibre, points are rendered by deck.gl.

#### Change List
- Match `GlobeViewport`'s scale, nearZ, farZ with maplibre's `GlobeTransform`
- `GlobeView` switch to `WebMercatorViewport` at zoom>=12. maplibre performs interpolation between the two projections at z [11, 12]. We may need to do the same, but the difference is honestly very subtle.
- Some golden images are updated because the zoom -> scale mapping has changed. The new implementation uses an adaptive scale that depends on the latitude of the viewport center.

### CODE REVIEW

**Review by felixpalmer — CHANGES_REQUESTED** [review]
Looks promising, but I get the page freezing after playing with the controls when viewing the full globe

**Comment by birkskyum** [comment]
Thank you for looking into this. It would be fantastic to have the deck layers in sync with the maplibre globe. Let me know if anything can be improved on the maplibre side for this. A note is that we're slowly making steps towards supporting terrain on the globe too - for deck layers I imagine that would i.e. be relevant for scenegraphlayer and more. It's flaky still, but there's a number of optimizations in the pipeline. I just wanted to keep you in the loop of where things are - there's a small prototype demo of it [here](https://stackblitz.com/edit/stackblitz-starters-sjqxyy?file=index.html).

File: test/modules/geo-layers/tileset-2d/utils.spec.ts [hunk 1]
```diff
@@ -275,7 +275,7 @@ const TEST_CASES = [
       viewState: {
         longitude: -6,
         latitude: 58,
-        zoom: 1.5
+        zoom: 1.5 + 0.7353406094252244 // Math.log2(1 / Math.PI / Math.cos(58 / 180 * Math.PI))
```
[the remark this answers is not in the document]
Comment (chrisgervang) [reply test/modules/geo-layers/tileset-2d/utils.spec.ts:278]: Does this constant help see us abnormalities in the images, or have some other benefit for testing?

File: test/render/test-cases/views.js [hunk 2]
```diff
@@ -101,9 +101,7 @@ export default [
       viewState: {
         longitude: -100,
         latitude: 80,
-        zoom: 0,
-        pitch: 0,
-        bearing: 0
+        zoom: -1
```
Comment (chrisgervang) [inline test/render/test-cases/views.js:104]: What's a negative zoom mean for globe?

File: test/modules/geo-layers/tileset-2d/utils.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibgreen) [reply test/modules/geo-layers/tileset-2d/utils.spec.ts:278]: Nice if comment explained what the formula is doing in a few words.

File: modules/core/src/viewports/globe-viewport.ts [hunk 3]
```diff
@@ -59,37 +62,39 @@ export type GlobeViewportOptions = {
...
+      farZMultiplier = 1,
       resolution = 10
     } = opts;
 
-    let {height, altitude = 1.5} = opts;
+    let {height, altitude = 1.5, fovy} = opts;
 
     height = height || 1;
-    altitude = Math.max(0.75, altitude);
+    if (fovy) {
```
Comment (ibgreen) [inline modules/core/src/viewports/globe-viewport.ts:82]: This aligns us more with other views, right?
We always support fovy and altitude is a backup for geospatial views?

File: modules/core/src/viewports/globe-viewport.ts [hunk 4]
```diff
@@ -59,37 +62,39 @@ export type GlobeViewportOptions = {
...
+      altitude = fovyToAltitude(fovy);
+    } else {
+      fovy = altitudeToFovy(altitude);
+    }
+    // Used to match globe and web mercator projection at high zoom
+    const scaleAdjust = 1 / Math.PI / Math.cos((latitude * Math.PI) / 180);
```
Comment (ibgreen) [inline modules/core/src/viewports/globe-viewport.ts:88]: Is this the new math logic? Almost missed it. Not sure we can put this in a separate method or something to make it stand out mode?
Commenting generously here would be appreciated.

**Review by ibgreen — APPROVED** [review]
This looks quite clean, I don't see any immediate reason why we wouldn't want to merge this as experimental in 9.1.

**Review by felixpalmer — APPROVED** [review]
@Pessimistress I believe the page freeze is unrelated to this PR https://github.com/visgl/deck.gl/issues/9265

File: modules/core/src/viewports/globe-viewport.ts [hunk 5]
```diff
@@ -59,37 +62,39 @@ export type GlobeViewportOptions = {
...
+  longitude!: number;
+  latitude!: number;
+  resolution!: number;
 
   constructor(opts: GlobeViewportOptions = {}) {
     const {
       latitude = 0,
       longitude = 0,
       zoom = 0,
-      nearZMultiplier = 0.1,
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/core/src/viewports/globe-viewport.ts:73]: Linked to maplibre source

File: modules/core/src/controllers/globe-controller.ts [hunk 6]
```diff
@@ -20,7 +23,7 @@ class GlobeState extends MapState {
     if (longitude < -180 || longitude > 180) {
       props.longitude = mod(longitude + 180, 360) - 180;
     }
-    props.latitude = clamp(latitude, -89, 89);
+    props.latitude = clamp(latitude, -MAX_VALID_LATITUDE, MAX_VALID_LATITUDE);
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/core/src/controllers/globe-controller.ts:26]: If we let latitude=90 then the controller goes into gimball lock. It is a solvable problem, but irrelevant to this PR.

File: modules/core/src/viewports/globe-viewport.ts (hunk 3 above)
Comment (Pessimistress) [reply modules/core/src/viewports/globe-viewport.ts:82]: That is correct.
`altitude` is a parameter we inherited from the mapbox/maplibre. The name is a bit misleading. It refers to the camera's distance to the sea plane.

**Pull request merged**